### ISSUE TITLE
Link rules?

### ISSUE THREAD

**Issue by badtant** [dequelabs/axe-core#209]
Hi,

I've created the following links and wanted to check if they are valid or not:
```
1. <a href=""></a>
2. <a href="">test2</a>
3. <a href="#">test3</a>
```

I only get an error on no 1. "Links must have discernible text". 
Are the other ones acceptable? I know that react-a11y and eslint-plugin-jsx-a11y check for no 3. Maybe that is something to include in axe too?

**Comment by marcysutton** [comment]
@dylanb I'd be curious to hear your take. When I tested this issue recently every screen reader I tested used the link text and not the URL, even in the rotor/links list. http://codepen.io/marcysutton/pen/AXaAyg

**Comment by dylanb** [comment]
Only anchor 1 is an accessibility problem. The other ones "may" be accessibility problems. For example if the JavaScript attached to those anchors is looking for anything other than a click event to do its magic, then the action might not be triggered using a screen reader or keyboard. There is no way that axe can tell that, so we don't flag it because it is most likely to be a false positive.

**Comment by badtant** [comment]
Ok, so no 2 is fine. But doesn't no 3 indicate that you should use a button instead?

**Comment by marcysutton** [comment]
As Dylan mentioned, it would likely be a false positive. Perhaps we could add a Best Practice rule instead, though.

**Comment by badtant** [comment]
Sounds good to me :)
